src/OrchardCore.Modules/OrchardCore.Autoroute/Handlers/AutorouteContentLocalizationHandler.cs

```diff
@@ -0,0 +1,21 @@
+using System.Threading.Tasks;
+using OrchardCore.Autoroute.Models;
+using OrchardCore.ContentManagement;
+
+namespace OrchardCore.ContentLocalization.Handlers;
+
+public class AutorouteContentLocalizationHandler : IContentLocalizationHandler
+{
+    public Task LocalizedAsync(LocalizationContentContext context) => Task.CompletedTask;
+
+    public Task LocalizingAsync(LocalizationContentContext context)
+    {
+        if (context.ContentItem.Has<AutoroutePart>())
+        {
+            // Clearing the AutoroutePart path to regenerate the permalink automatically.
+            context.ContentItem.Content.AutoroutePart.Path = null;
```

Comment:
Do this statically typed instead, with `Alter()`.

Reply:
We could :)

Reply:
Why the plural, how else is there Hisham?! :D